docs/user-guide/projects.md

```diff
@@ -124,27 +115,37 @@ argocd proj deny-namespace-resource <PROJECT> <GROUP> <KIND>
 
 ### Assign Application To A Project
 
-The application project can be changed using `app set` command. In order to change the project of
-an app, the user must have permissions to access the new project.
+The application project can be changed using `app set` command. In order to change the project of an app, the user must have permissions to access the new project.
 
 ```
 argocd app set guestbook-default --project myproject
 ```
 
 ## Project Roles
 
-Projects include a feature called roles that enable automated access to a project's applications.
-These can be used to give a CI pipeline a restricted set of permissions. For example, a CI system
-may only be able to sync a single app (but not change its source or destination).
+Projects include a feature called roles that can be used to determine who and what can be done applications associated with the project. As an example, it can be used to give a CI pipeline a restricted set of permissions allowing sync operations on a single app (but not change its source or destination).
+
+Projects can have multiple roles, and those roles can have different access granted to them. These permissions are called policies, and they are stored within the role as a list of policy strings. A role's policy can only grant access to that role and are limited to applications within the role's project. However, the policies have an option for granting wildcard access to any application within a project.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AppProject
+metadata:
+  name: sample-test-project
+spec:
+  ...
+  roles:
+  - description: some-role
+    groups:
+    - some-user
+    name: admin
+    policies:
+    - p, proj:sample-test-project:some-role, applications, *, *, allow
```

Comment:
I think these _might_ also be manageable: `applicationsets`, `clusters`, `repositories`, `exec`, `logs`, `extensions`. But I haven't confirmed.

Reply:
yes.. they are all available just like in the main argocd-rbac-cm definitions. This was just an example. Let me add a sentence to clarify that.

Reply:
done